### ISSUE TITLE
Error: failed to get state for index 0

### ISSUE THREAD

**Issue by lalithsuresh** [earthly/earthly#3150]
**What went wrong?**

We've been regularly running into the following exception. We're not sure what's triggering it. It seems sporadic and the only way to avoid it is to prune the cache with `earthly prune --reset`. The error looks like this:

```
Error: build target: build main: failed to solve: async force execution for +prepare-cache: unlazy force execution: failed to compute cache key: failed to get state for index 0 on 
```
Here's a recent run where it happened: https://github.com/feldera/dbsp/actions/runs/5675940533/job/15381943181

In this particular instance, rerunning the pipeline did not help. So I had to prune the cache to get things working again.

I haven't found a clear pattern that would help me reproduce it yet. It happens sporadically. 

I don't have a minimal example handy, but this is the Earthfile we're using: https://github.com/feldera/dbsp/blob/main/Earthfile

**What should have happened?**

The error shouldn't be happening given that it is sporadic.

Pruning the cache seems to help.

**Other Helpful Information**

* Version info: earthly version v0.7.0 cb2217400afdce4b49221dfa2eb64c06373141ce linux/amd64; Ubuntu 22.04.2 LTS (Jammy Jellyfish)

**Comment by alexcb** [comment]
This might possibly relate to #2957

**Comment by ingwarsw** [comment]
We also started to encounter that issue really often.

```
Error: build target: build main: failed to solve: async force execution for +kbt-build-all: unlazy force execution: failed to compute cache key: failed to get state for index 1 on [eyJpdHJubCI6dHJ1ZX0=] (fakecopy2) +kbt-build-all depends on +kbt-push
```

It started to happen when I added `--push` to one of curl commands thats pushing debian package to repository.
Without it I didnt saw it happening.
With push we have like 30% of builds failing..

I will try to create reproduction case, maybe that would help pinpointing it.

**Comment by alexcb** [comment]
#2957 has been fixed as of of [v0.7.21](https://github.com/earthly/earthly/releases/tag/v0.7.21); that issue was a race condition inside the buildkit scheduler, in particular when merging (deduplicating) multiple edges within the build graph. I _think_ this issue is related, but triggered a different error message. I'm going to close this as fixed, but if it happens again please reopen this.

**Comment by zapling** [comment]
I also run into this issue after quite a few invocations when testing on `earthly-linux-amd64 version v0.8.14 6b41f8409d7ffef0d25072c2c04250074b6e3c72 linux/amd64; Arch Linux` .

```
Error: unlazy force execution: failed to compute cache key: failed to get state for index 1 on [eyJpdHJubCI6dHJ1ZX0=] (fakecopy2) ertia+deploy-test depends on .ertia+build
```

Was resolved after the `earthly prune --reset` command. I was also using the `--no-cache` option.
